clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/model/EcsInstance.java

```diff
@@ -0,0 +1,82 @@
+package com.netflix.spinnaker.clouddriver.ecs.model;
+
+import com.amazonaws.services.ec2.model.InstanceStatus;
+import com.amazonaws.services.ecs.model.Task;
+import com.netflix.spinnaker.clouddriver.ecs.EcsCloudProvider;
+import com.netflix.spinnaker.clouddriver.model.HealthState;
+import com.netflix.spinnaker.clouddriver.model.Instance;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Map;
+
+public class EcsInstance implements Instance, Serializable {
+  private String name;
+  private HealthState healthState;
+  private Long launchTime;
+  private String zone;
+  private List<Map<String, String>> health;
+  private String providerType;
+  private String cloudProvider;
+
+  public EcsInstance(String name, Task task, InstanceStatus ec2Instance) {
```

Comment:
It feels EcsInstance needs to know a lot about other classes in order to work itself.  Constructors in Java are usually pretty dumb 

From a testing perspective, we'd have to know a lot about Task and InstanceStatus to properly test this EcsInstance class, and it would create dependencies that aren't no-brainers to maintain

Reply:
Following the example of similar constructors:

https://github.com/lookout/clouddriver/blob/master/clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/model/KubernetesInstance.groovy#L55

For testing purposes, in python I would just create a mock Task and InstanceStatus classes and feed them to the class. I am not sure how we would approach this in Java.

If we remove this dependency, we would just shift the burden of knowing how to map the Task and Instance Status field to the EcsInstance (EcsTask) fields to the caller, which seems a bit wonky.. specially if there is more than one caller